core/src/main/java/edu/wpi/grip/core/operations/composite/SaveImageOperation.java

```diff
@@ -0,0 +1,109 @@
+package edu.wpi.grip.core.operations.composite;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableList;
+import edu.wpi.grip.core.FileManager;
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.OperationDescription;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.sockets.SocketHint;
+import edu.wpi.grip.core.sockets.SocketHints;
+import edu.wpi.grip.core.util.Icon;
+import org.bytedeco.javacpp.BytePointer;
+import org.bytedeco.javacpp.IntPointer;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.bytedeco.javacpp.opencv_core.Mat;
+import static org.bytedeco.javacpp.opencv_imgcodecs.CV_IMWRITE_JPEG_QUALITY;
+import static org.bytedeco.javacpp.opencv_imgcodecs.imencode;
+
+/**
+ * Save JPEG files periodically to the local disk.
+ */
+public class SaveImageOperation implements Operation {
+
+    public static final OperationDescription DESCRIPTION =
+            OperationDescription.builder()
+                    .name("Save Images to Disk")
+                    .summary("Save image periodically to local disk")
+                    .category(OperationDescription.Category.MISCELLANEOUS)
+                    .icon(Icon.iconStream("publish-video"))
+                    .build();
+
+    private final SocketHint<Mat> inputHint = SocketHints.Inputs.createMatSocketHint("Input", false);
+    private final SocketHint<Number> qualityHint = SocketHints.Inputs.createNumberSliderSocketHint("Quality", 90, 0, 100);
+    private final SocketHint<Number> periodHint = SocketHints.Inputs.createNumberSpinnerSocketHint("Period", 0.1);
+    private final SocketHint<Boolean> activeHint = SocketHints.Inputs.createCheckboxSocketHint("Active", false);
+
+    private final SocketHint<Mat> outputHint = SocketHints.Outputs.createMatSocketHint("Output");
+
+    private final InputSocket<Mat> inputSocket;
+    private final InputSocket<Number> qualitySocket;
+    private final InputSocket<Number> periodSocket;
+    private final InputSocket<Boolean> activeSocket;
+
+    private final OutputSocket<Mat> outputSocket;
+
+    private final FileManager fileManager;
+    private final BytePointer imagePointer = new BytePointer();
+    private Stopwatch stopwatch = Stopwatch.createStarted();
+    private SimpleDateFormat format = new SimpleDateFormat("yyyyMMdd_HHmmss.SSS");
```

Comment:
`java.util.Date` is deprecated, so use `java.time.format.DateTimeFormatter` instead, e.g.
`private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyMMdd_HHmmss.SSS");`

That format also looks a little wonky (I"m seeing stuff like `20160608_224640.606`).